src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -289,42 +349,95 @@ public boolean isPrivateIpUsed() {
         return privateIpUsed;
     }
 
+    @DataBoundSetter
+    public void setPrivateIpUsed(boolean privateIpUsed) {
+        this.privateIpUsed = privateIpUsed;
+    }
+
     public boolean isAlwaysReconnect() {
         return alwaysReconnect;
     }
 
+    @DataBoundSetter
+    public void setAlwaysReconnect(boolean alwaysReconnect) {
+        this.alwaysReconnect = alwaysReconnect;
+    }
+
     public String getLabelString() {
         return labelString;
     }
 
     public int getIdleMinutes() {
-        return (idleMinutes != null) ? idleMinutes : 0;
+        return idleMinutes;
+    }
+
+    @DataBoundSetter
+    public void setIdleMinutes(Integer idleMinutes) {
+        this.idleMinutes = (idleMinutes != null) ? idleMinutes : 0;
     }
 
     public int getMaxSize() {
         return maxSize;
     }
 
+    @DataBoundSetter
+    public void setMaxSize(int maxSize) {
+        if (maxSize < minSize) {
+            int newMaxSize = minSize > 0 ? minSize : 1;
+            warning("Cloud parameter 'maxSize' can't be less than 'minSize' or 1, setting to %d", newMaxSize);
+            maxSize = newMaxSize;
+        }
+        this.maxSize = Math.max(1, maxSize);
+    }
+
     public int getMinSize() {
         return minSize;
     }
 
-    public int getMinSpareSize() {
+    @DataBoundSetter
+    public void setMinSize(int minSize) {
+        if (minSize < 0) {
+            warning("Cloud parameter 'minSize' can't be less than 0, setting to 0");
+        }
+        this.minSize = Math.max(0, minSize);
+    }
+
+    public synchronized int getMinSpareSize() {
         return minSpareSize;
     }
 
+    @DataBoundSetter
+    public synchronized void setMinSpareSize(int minSpareSize) {
+        this.minSpareSize = Math.max(0, minSpareSize);
+    }
+
     public int getMaxTotalUses() {
         return maxTotalUses;
     }
 
+    @DataBoundSetter
+    public void setMaxTotalUses(String maxTotalUses) {
+        this.maxTotalUses = StringUtils.isBlank(maxTotalUses) ? DEFAULT_MAX_TOTAL_USES : Integer.parseInt(maxTotalUses);
```

Comment:
Since the default is already set.

```
if (StringUtils.isNotBlank(maxtotalUses)) {
     this.maxTotalUses = Integer.parseInt(maxTotalUses);
}

```